cpp/dolfinx/fem/Function.h

```diff
@@ -195,13 +210,29 @@ class Function
     fem::interpolate(*this, fx, cells);
   }
 
-  /// Interpolate an Expression (based on ufl)
-  /// @param[in] expr The function to be interpolated.
-  /// @param[in] cells The cells (local to process) to interpolate into
-  void interpolate(const Expression<T>& expr,
+  /// Interpolate an expression function on the whole domain
+  /// @param[in] f The expression to be interpolated
+  void interpolate(
+      const std::function<xt::xarray<T>(const xt::xtensor<double, 2>&)>& f)
+  {
+    assert(_function_space);
+    assert(_function_space->mesh());
+    const int tdim = _function_space->mesh()->topology().dim();
+    auto cell_map = _function_space->mesh()->topology().index_map(tdim);
+    assert(cell_map);
+    std::int32_t num_cells = cell_map->size_local() + cell_map->num_ghosts();
+    std::vector<std::int32_t> cells(num_cells, 0);
+    std::iota(cells.begin(), cells.end(), 0);
+    interpolate(f, cells);
+  }
+
+  /// Interpolate an Expression (based on UFL)
+  /// @param[in] e The function to be interpolated
+  /// @param[in] cells The cells to interpolate into
+  void interpolate(const Expression<T>& e,
```

Comment:
We should have an interpolate for `Expression` over the whole domain as well I guess.